File: Unit-2-Lesson-04/externalRGB.c
// Created by Caleb Andreas
// Created on March 3, 2026
// This program goes through the colours of an external RGB LED.

// Variables.
int pin5 = 5; // Set red output to pin5.
int pin6 = 6; // Set green output to pin6.
int pin7 = 7; // Set blue output to pin7.

void setup() {
  // Initialize digital pin LED_BUILTING as an output.
  pinMode(pin5, OUTPUT);
  pinMode(pin6, OUTPUT);
  pinMode(pin7, OUTPUT);
}

void loop() {
  digitalWrite(pin5, HIGH);  // Turn red on.
  digitalWrite(pin6, LOW);   // Turn green off.
  digitalWrite(pin7, LOW);   // Turn blue off.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, LOW);   // Turn red off.
  digitalWrite(pin6, HIGH);  // Turn green on.
  digitalWrite(pin7, LOW);   // Turn blue off.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, LOW);   // Turn red off.
  digitalWrite(pin6, LOW);   // Turn green off.
  digitalWrite(pin7, HIGH);  // Turn blue on.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, HIGH);  // Turn red on.
  digitalWrite(pin6, HIGH);  // Turn green on.
  digitalWrite(pin7, LOW);   // Turn blue off.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, LOW);   // Turn red off.
  digitalWrite(pin6, HIGH);  // Turn green on.
  digitalWrite(pin7, HIGH);  // Turn blue on.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, HIGH);  // Turn red on.
  digitalWrite(pin6, LOW);   // Turn green off.
  digitalWrite(pin7, HIGH);  // Turn blue on.
  delay(1000);               // Wait for a second.
  
  digitalWrite(pin5, HIGH);  // Turn red on.
  digitalWrite(pin6, HIGH);  // Turn green on.
  digitalWrite(pin7, HIGH);  // Turn blue on.
  delay(1000);               // Wait for a second.
}

File: Unit-2-Lesson-04/external_RGB.py
#!/usr/bin/env python3
"""
Created by: Caleb Andreas
Created on: February 24, 2026
This program is a program for the pi pico that goes through the colours of an external RGB LED.
"""

import board
import digitalio
import time

# Setup.
pin5 = digitalio.DigitalInOut(board.GP5) # Set red output to pin5.
pin6 = digitalio.DigitalInOut(board.GP6) # Set green output to pin6.
pin7 = digitalio.DigitalInOut(board.GP7) # Set blue output to pin7.
pin5.direction = digitalio.Direction.OUTPUT
pin6.direction = digitalio.Direction.OUTPUT
pin7.direction = digitalio.Direction.OUTPUT


while True:
    pin5.value = True   # Turn red on.
    pin6.value = False  # Turn green off.
    pin7.value = False  # Turn blue off.
    time.sleep(1)       # Wait for a second.

    pin5.value = False  # Turn red off.
    pin6.value = True   # Turn green on.
    pin7.value = False  # Turn blue off.
    time.sleep(1)       # Wait for a second.

    pin5.value = False  # Turn red off.
    pin6.value = False  # Turn green off.
    pin7.value = True   # Turn blue on.
    time.sleep(1)       # Wait for a second.

    pin5.value = True   # Turn red on.
    pin6.value = True   # Turn green on.
    pin7.value = False  # Turn blue off.
    time.sleep(1)       # Wait for a second.

    pin5.value = False  # Turn red off.
    pin6.value = True   # Turn green on.
    pin7.value = True   # Turn blue on.
    time.sleep(1)       # Wait for a second.

    pin5.value = True   # Turn red on.
    pin6.value = False  # Turn green off.
    pin7.value = True   # Turn blue on.
    time.sleep(1)       # Wait for a second.

    pin5.value = True   # Turn red on.
    pin6.value = True   # Turn green on.
    pin7.value = True   # Turn blue on.
    time.sleep(1)       # Wait for a second.
